### PR TITLE
useradd: doesn't generate /var/spool/mail/$USER with the proper SELinux user identity

### PR DESCRIPTION
**Pull request by ikerexxe**
Explanation: use set_selinux_file_context() and reset_selinux_file_context() for create_mail() just as is done for create_home()

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1690527

### CODE REVIEW

**Comment by pbrezina** [comment]
I did not try it but code looks good to me.

File: src/useradd.c [hunk 1]
```diff
@@ -2210,6 +2211,15 @@ static void create_mail (void)
 			sprintf (file, "%s/%s/%s", prefix, spool, user_name);
 		else
 			sprintf (file, "%s/%s", spool, user_name);
+
+#ifdef WITH_SELINUX
+		if (set_selinux_file_context (spool) != 0) {
```
Comment (cgzones) [inline src/useradd.c]: why are you using the path of `spool` and not `file`?

Comment (ikerexxe) [reply src/useradd.c]: I agree that this may not be a good idea from a security point of view. So, I'm changing the fix to change directly the file.

**Comment by cgzones** [comment]
I do not think this is going to work: now you are just copying the context from `/var/spool/mail` to `/var/spool/mail/joe`.
These do not have the same context:
  - `/var/spool/mail` has usually the context: `system_u:object_r:mail_spool_t`
  - `/var/spool/mail/joe` has a user-identity modified context: `user_u:object_r:mail_spool_t`

I think the first version was quite good (just change `set_selinux_file_context (spool)` to `set_selinux_file_context (file)`).

**Comment by ikerexxe** [comment]
But isn't this approach a little bit dangerous? It opens the gate for another process to create a file in this folder with that context.

**Comment by ikerexxe** [comment]
What would happen if between the context set and the context reset another process creates another file in this same folder? Would this file have "unconfined_u" or "system_u" as the SELinux user identity? Would this pose a security risk?

**Comment by ikerexxe** [comment]
I'm sorry but I got confused so please forget about my messages. I will proceed to fix this problem as you proposed.

**Comment by bachradsusi** [comment]
Before you merge it, please consider a different commit message

> useradd: doesn't generate /var/spool/mail/$USER with the proper SELinux user identity

In fact the patch *does* generate /var/spool/mail/$USER with the proper SELinux label based on SELinux user identity

**Comment by ikerexxe** [comment]
> Before you merge it, please consider a different commit message
> 
> > useradd: doesn't generate /var/spool/mail/$USER with the proper SELinux user identity
> 
> In fact the patch _does_ generate /var/spool/mail/$USER with the proper SELinux label based on SELinux user identity

Done!

**Comment by hallyn** [comment]
Thanks, all.

**Pull request merged**